src/components/header/share-modal/renderer.tsx

```diff
@@ -63,55 +81,213 @@ class ShareModal extends React.PureComponent<Props, State> {
     this.exportURL();
   }
 
+  public updatePrivacy(event) {
+    this.setState({
+      gistPrivate: event.target.checked
+    });
+  }
+
+  public fileNameChange(event) {
+    this.setState({
+      gistFileName: event.target.value
+    });
+  }
+
+  public titleChange(event) {
+    this.setState({
+      gistTitle: event.target.value
+    });
+  }
+
+  public createGist() {
+    this.setState({
+      done: false
+    });
+    const body = {
+      content: this.props.editorString,
+      name: this.state.gistFileName || "spec",
+      title: this.state.gistTitle,
+      privacy: this.state.gistPrivate
+    };
+    const cookieValue = encodeURIComponent(getCookie(COOKIE_NAME));
+    fetch(`${BACKEND_URL}gists/create`, {
+      body: JSON.stringify(body),
+      mode: "cors",
+      credentials: "include",
+      headers: {
+        "Content-Type": "application/json",
+        Cookie: `${COOKIE_NAME}=${cookieValue}`
+      },
+      method: "post"
+    })
+      .then(res => {
+        return res.json();
```

Comment:
This line is throwing `SyntaxError: Unexpected token C in JSON at position 0`. Here is the answer I referred to solve this problem: https://stackoverflow.com/a/43094587/11751642. But this returns `{}` when user is not logged in which shouldn't be the case since it return `{"handle":"","isAuthenticated":false,"name":"","profilePicUrl":""}`
`

Reply:
Is this fixed? Sounds like we are not requesting a JSON.

Reply:
@domoritz I couldn't exactly get why the error persisted though